File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;

        int[] leftProduct = new int[n];
        int[] rightProduct = new int[n];

        for(int i = 0; i<n; i++){
            if(i==0){
                leftProduct[0] = nums[i];
                rightProduct[n-1] = nums[n-1];
            }
            else{
                leftProduct[i] = nums[i]*leftProduct[i-1];
                rightProduct[n-1-i] = nums[n-1-i]*rightProduct[n-i];
            }
        }

        int[] ans = new int[n];

        for(int i = 0; i<n; i++){
            if(i==0){
                ans[i] = rightProduct[1];
            }
            else if(i==n-1){
                ans[i] = leftProduct[i-1];
            }
            else{
                ans[i] = leftProduct[i-1]*rightProduct[i+1];
            }
        }

        return ans;
    }
}  

File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
public:
    void getCombinationSum(vector<int>& nums, int target, int curr, int currSum, vector<int>& v, vector<vector<int>>& ans){
        if(currSum>target || curr>=nums.size()){
            return;
        }

        if(currSum==target){
            ans.push_back(v);
            return;
        }

        v.push_back(nums[curr]);
        currSum+=nums[curr];

        getCombinationSum(nums, target, curr, currSum, v, ans);

        v.pop_back();

        currSum-=nums[curr];

        getCombinationSum(nums, target, curr+1, currSum, v, ans);
        
        return;

    }

    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        vector<vector<int>> ans;
        vector<int> v; 

        getCombinationSum(nums, target, 0, 0, v, ans);
        return ans;
    }
};
